test/TesterInternal/AssemblyLoaderTests.cs

```diff
@@ -10,163 +10,39 @@
-                    };
-            var excludeCriteria =
-                new AssemblyLoaderPathNameCriterion[]
-                    {
-                        AssemblyLoaderCriteria.ExcludeResourceAssemblies,
-                        AssemblyLoaderCriteria.ExcludeFileNames(exclusionList)
-                    };
-            var loadProvidersCriteria =
-                new AssemblyLoaderReflectionCriterion[]
-                    {
-                        AssemblyLoaderCriteria.LoadTypesAssignableFrom(typeof(IProvider))
-                    };
-
-            return AssemblyLoader.NewAssemblyLoader(directories, excludeCriteria, loadProvidersCriteria, logger);
-        }
-
-        private void DiscoverAssemblies(AssemblyLoader loader, List<string> exclusionList, bool validate = true)
+        [Fact, TestCategory("AssemblyLoader"), TestCategory("BVT"), TestCategory("Functional")]
+        public void AssemblyLoaderTest()
         {
-            var result = loader.DiscoverAssemblies();
-
-            var text = new StringBuilder();
-            text.Append("\nFound assemblies:");
-            foreach (var i in result)
-                text.Append(String.Format("\n\t* {0}", i));
-            logger.Info(text.ToString());
-
-            if (validate)
-            {
-                var found = false;
-                foreach (var i in result)
-                {
-                    var fileName = Path.GetFileName(i);
-                    // we shouldn't have any blacklisted assemblies in the list.
-                    Assert.False(exclusionList.Contains(fileName), "Assemblies on an exclusion list should be ignored.");     
-                    if (fileName == ExpectedFileName)
-                        found = true;
-                }
-                Assert.True(
-                    found, 
-                    String.Format(
-                        "{0} should have been found by the assembly loader", 
-                        ExpectedFileName));                
-            }
+            var catalog = NewCatalog();
+            var loader = AssemblyLoader.NewAssemblyLoader(catalog);
         }
 
-        private void ExpectException(Action action)
+        private AssemblyCatalog NewCatalog()
         {
-            try
-            {
-                action();
-            }
-            catch (AggregateException e)
-            {
-                if (e.InnerExceptions.Count != 2 || 
-                    e.InnerExceptions[0].Message != "Inner Exception #1" ||
-                    e.InnerExceptions[1].Message != "Inner Exception #2")
-                {
-                    throw;
-                }
-            }
+            return new AssemblyCatalog()
+                .WithAssembly(@".\TestGrainInterfaces.dll")
+                .WithAssembly(@".\TestGrains.dll")
+                .WithAssembly(@".\TestInternalGrainInterfaces.dll")
+                .WithAssembly(@".\TestInternalGrains.dll");
```

Comment:
I thought you were using assembly names, not paths to the DLLs. Nevertheless I see here and in many config files that you are passing paths